jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -117,16 +112,17 @@ public Callback getCallback()
 
     /**
      * Associate this channel with a specific request.
-     * This is called by the ServletContextHandler when a core {@link Request} is accepted and associated with
-     * a servlet mapping.
+     * This method is called by the {@link ServletContextHandler} when a core {@link Request} is accepted and associated with
+     * a servlet mapping. The association remains until {@link #recycle()} is called.
      * @param servletContextRequest The servlet context request to associate
      * @see #recycle()
      */
     public void associate(ServletContextRequest servletContextRequest)
     {
-        _state.recycle();
+        // TODO This is needed as requests that are handled before the ServletHandler are not recycled.
```

Comment:
Oh the irony - the comment on the PR says its purpose is to remove TODOs, and here's a new one being added! :grin:

Reply:
Oh it didn't need the TODO part in the comment :)